sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1813,6 +1813,17 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("global temp db is used as a table name under v2 catalog") {
+    val globalTempDB = spark.sessionState.conf.getConf(StaticSQLConf.GLOBAL_TEMP_DATABASE)
+    val t = s"testcat.$globalTempDB"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id bigint, data string) USING foo")
+      sql("USE testcat")
+      // The following should not throw AnalysisException, but should use `testcat.$globalTempDB`.
+      sql(s"DESCRIBE TABLE $globalTempDB")
```

Comment:
what happens if we have a table `testcat` under catalog `testcat`, and then run `DESCRIBE TABLE testcat`?

Reply:
good catch! I think `CatalogAndIdentifierParts` should do something similar to `CatalogAndIdentifier` or integrate the `globalTempDB` logic into `CatalogAndIdentifier` and remove `CatalogAndIdentifierParts`